compiler/optimizer/OMRTransformUtil.cpp

```diff
@@ -406,10 +406,15 @@ OMR::TransformUtil::createConditionalAlternatePath(TR::Compilation* comp,
    {
    cfg->setStructure(0);
 
+   bool mergeToElseBlock = elseBlock == mergeBlock ? true : false;
+
    TR::Block* ifBlock = elseBlock;
    ifBlock->prepend(ifTree);
    elseBlock = ifBlock->split(ifTree->getNextTreeTop(), cfg, false /*fixupCommoning*/, true /*copyExceptionSuccessors*/);
 
+   if (mergeToElseBlock)
```

Comment:
Is there a reason for using a dedicated variable instead of just:

```suggestion
   if (elseBlock == mergeBlock)
```

Also, given that `elseBlock` and `mergeBlock` are pointers, if `elseBlock == mergeBlock` then isn't `mergeBlock = elseBlock` a no-op?

Reply:
`elseBlock` is changed at line 413. `split` creates a new block and assign it to `elseBlock`.

Reply:
I see. Thank you for clarifying that. :slightly_smiling_face: